pkg/cloud-init/cloud-init_test.go

```diff
@@ -204,6 +204,74 @@ var _ = Describe("CloudInit", func() {
 					verifyCloudInitIso(cloudInitData)
 				})
 			})
+			Context("with cloudInitNoCloud userData and networkData without resolvData", func() {
+				It("should success", func() {
+					userData := "fake\nuser\ndata\n"
+					networkData := "fakeData\n" + cloudInitDelimiter
+					cloudInitData := &v1.CloudInitNoCloudSource{
+						UserData:    userData,
+						NetworkData: networkData,
+					}
+					verifyCloudInitIso(cloudInitData)
+				})
+			})
+			Context("with cloudInitNoCloud userData and networkData without resolvData or delimiter", func() {
+				It("should success", func() {
+					userData := "fake\nuser\ndata\n"
+					networkData := "fakeData\n"
+					cloudInitData := &v1.CloudInitNoCloudSource{
+						UserData:    userData,
+						NetworkData: networkData,
+					}
+					verifyCloudInitIso(cloudInitData)
+				})
+			})
+			Context("with cloudInitNoCloud userData and networkData with resolvData", func() {
+				It("should success", func() {
+					userData := "#cloud-config\nfakeData\n"
+					networkData := "fakeData\n" + cloudInitDelimiter + "fakeResolv"
+					cloudInitData := &v1.CloudInitNoCloudSource{
+						UserData:    userData,
+						NetworkData: networkData,
+					}
+					verifyCloudInitIso(cloudInitData)
+				})
+			})
+			Context("with cloudInitNoCloud userData and networkData with resolvData and manage_resolv_conf:", func() {
+				It("should success", func() {
+					userData := "#cloud-config\nmanage_resolv_conf: true\n"
+					networkData := "#cloud-config\nmanage_resolv_conf: true\n" + cloudInitDelimiter + "fakeResolv"
+					cloudInitData := &v1.CloudInitNoCloudSource{
+						UserData:    userData,
+						NetworkData: networkData,
+					}
+					verifyCloudInitIso(cloudInitData)
+				})
+			})
+			Context("with cloudInitNoCloud userData and networkData with resolvData without #cloud-config", func() {
+				It("should success", func() {
+					userData := "fake\nuser\ndata\n"
+					networkData := "fakeNetwork\n" + cloudInitDelimiter + "fakeResolv"
+					cloudInitData := &v1.CloudInitNoCloudSource{
+						UserData:    userData,
+						NetworkData: networkData,
+					}
+					verifyCloudInitIso(cloudInitData)
+				})
+			})
+			Context("with cloudInitNoCloud networkData source", func() {
```

Comment:
cloud-init requires user-data even if it just contains a single line with "#cloud-config". At least this was the behavior in my testing.

Reply:
So maybe we can if the user-data is not configure but the network-data is just add `#cloud-config` to the user-data or just use the network-data as the only configuration?

Reply:
I've thought about it, forgot why I decided not to do it this first go around.

Reply:
So I think I was worried about order of operations for data from secrets(I don't think it is an issue now). It would require some changes to api validation. I think this can be handled in a future issue/pr if someone actually needs this functionality.